tests/scattering2d/test_numpy_scattering2d.py

```diff
@@ -137,30 +137,29 @@ def test_fft(self, backend):
                       [x[0, 0] + x[0, 1] - x[1, 0] - x[1, 1],
                        x[0, 0] - x[0, 1] - x[1, 0] + x[1, 1]]])
 
-        z = backend.fft(x, direction='C2C')
 
-        assert np.allclose(y, z)
-
-        z = backend.fft(x, direction='C2C', inverse=True)
+        z = backend.ifft(x)
 
         z = z * 4
 
         assert np.allclose(y, z)
 
-        z = backend.fft(x, direction='C2R', inverse=True)
+        z = backend.irfft(x)
 
         z = z * 4
 
         assert not np.iscomplexobj(z)
-        assert np.allclose(np.real(y), z)
+        assert np.allclose(y.real, z)
 
+        x = np.random.randn(2, 2)
 
-    @pytest.mark.parametrize('backend', backends)
-    def test_fft_exceptions(self, backend):
-        with pytest.raises(RuntimeError) as record:
-            backend.fft(np.empty((2, 2)), direction='C2R',
-                        inverse=False)
-        assert 'done with an inverse' in record.value.args[0]
+        y = np.array([[x[0, 0] + x[0, 1] + x[1, 0] + x[1, 1],
+                       x[0, 0] - x[0, 1] + x[1, 0] - x[1, 1]],
+                      [x[0, 0] + x[0, 1] - x[1, 0] - x[1, 1],
+                       x[0, 0] - x[0, 1] - x[1, 0] + x[1, 1]]])
+
+        z = backend.rfft(x)
+        assert np.allclose(y.real, z)
```

Comment:
Why take the real part here? Also the sequence here is confusing. Why not take the `rfft` of a real signal, make sure it's correct, then compute the `ifft` and `irfft` of that transform and make sure those are correct?

Reply:
so make sure `ifft` and `irfft` both match?

Reply:
Yes. Only difference should be that the imaginary part for `ifft` should be zero.